### PR TITLE
Binary Exponentiation Algorithm.cpp

### PR DESCRIPTION
**Pull request by Shubh08am**
I have explained in detailed about Binary Exponentiation Algorithm which is used frequently in cp.

## 🛠️ Issue (Number)

Issue no #154
Issue no #175 
Issue no #173

## 👨‍💻 Changes proposed
Added Binary Exponentiation Algorithm in C++ Section inside Maths Section.

## ✔️ Check List (Check all the applicable boxes) 

- [ ✔️] My code follows the code style of this project.
- [✔️ ] This PR does not contain plagiarized content.
- [ ✔️] The title of my pull request is a short description of the requested changes.

- **Execution Screenshots**

imageimage

Issue no #175 
Sliding-Windows Algorithm 
## 👨‍💻 Changes proposed
Added questions related to Sliding-Windows Algorithm  in C++ Section inside Sliding-Windows Section.

## ✔️ Check List (Check all the applicable boxes) 

- [ ✔️] My code follows the code style of this project.
- [✔️ ] This PR does not contain plagiarized content.
- [ ✔️] The title of my pull request is a short description of the requested changes.

Issue no #173
Sieve of Eratosthenes Algorithm
## 👨‍💻 Changes proposed
Added Sieve of Eratosthenes Algorithm in C++ Section inside Maths Section.

## ✔️ Check List (Check all the applicable boxes) 

- [ ✔️] My code follows the code style of this project.
- [✔️ ] This PR does not contain plagiarized content.
- [ ✔️] The title of my pull request is a short description of the requested changes.

- **Execution Screenshots**

image
image

### CODE REVIEW

**Comment by Kumar-laxmi** [comment]
@Shubh08am Please resolve your **Codefactor issues**

File: C++/Sliding Window/README.md [hunk 1]
```diff
@@ -1 +1,3 @@
-# Sliding Window Algorithm
\ No newline at end of file
+# Sliding Window Algorithm
+
+This folder contains all the important questions related to sliding windows Algorithm.
```
Comment (Kumar-laxmi) [inline C++/Sliding Window/README.md:3]: Please add a table in this readme featuring all problems and their links

**Review by Kumar-laxmi — CHANGES_REQUESTED** [review]

**Comment by Kumar-laxmi** [comment]
@Shubh08am Please resolve your Codefactor issues

**Comment by Shubh08am** [comment]
> @Shubh08am Please resolve your Codefactor issues

Please can u explain me how to do that I'm not able to understand it.

**Comment by Kumar-laxmi** [comment]
@Shubh08am I think you are having redundant blank line at the end of file in every program just remove and see

**Comment by Shubh08am** [comment]
> @Shubh08am I think you are having redundant blank line at the end of file in every program just remove and see

@Kumar-laxmi done the changes.

**Review by Kumar-laxmi — APPROVED** [review]

**Pull request merged**